FinniversKit/Sources/SwiftUI Components/Button/SwiftUIFloatingButton.swift

```diff
@@ -0,0 +1,92 @@
+import SwiftUI
+
+public struct SwiftUIFloatingButton: View {
+    @ObservedObject var viewModel: SwiftUIFloatingButtonModel
+    @State private var isPressed: Bool = false
+    private let style: SwiftUIFloatingButtonStyle
+    private let action: () -> Void
+
+    public init(
+        style: SwiftUIFloatingButtonStyle = .default,
+        viewModel: SwiftUIFloatingButtonModel,
+        action: @escaping () -> Void
+    ) {
+        self.style = style
+        self.viewModel = viewModel
+        self.action = action
+    }
+
+    public var body: some View {
+        SwiftUI.Button(action: {
+            action()
+            isPressed = false
+        }, label: {
+            HStack {
+                viewModel.image
+                    .foregroundColor(style.tintColor)
+                if let title = viewModel.title {
+                    Text(title)
+                        .finnFont(.detail)
+                        .foregroundColor(style.titleColor)
+                }
+            }
+            .padding(.spacingM)
+            .background(isPressed ? style.highlightedBackgroundColor : style.primaryBackgroundColor)
+            .clipShape(Capsule())
+            .shadow(
+                color: style.shadowColor,
+                radius: style.shadowRadius,
+                x: style.shadowOffset.width,
+                y: style.shadowOffset.height
+            )
+        })
+        .buttonStyle(PlainButtonStyle())
+        .simultaneousGesture(LongPressGesture().onChanged { _ in
```

Comment:
Why is the simultaneous gesture modifier needed instead of a regular tap gesture?

Reply:
Dumb way of making sure the color stays changed when button is held. Better solved with a button style like in the above comment.